### PR TITLE
Add the require-skill and unrequire-skill commands

### LINKED ISSUE #105: [Logic] Commands require-skill and unrequire-skill, and add skill to create-group

### PR DESCRIPTION
**Pull request by ivan24004**
Closes #105

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        requireNonNull(args);
+        ArgumentMultimap argMultiMap = ArgumentTokenizer.tokenize(args, PREFIX_SKILL);
+
+        try {
+            Name name = ParserUtil.parseName(argMultiMap.getPreamble());
+            Set<Skill> skillSet =
+                    ParserUtil.parseSkills(argMultiMap.getAllValues(PREFIX_SKILL));
+
+            // will get caught by the catch clause, leave empty.
+            if (skillSet.isEmpty()) {
```
Comment (yuhinarita) [inline src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java:34]: I think its better to let parseSkills() throw a ParserException, than having an if-statement throw empty exception.
Edit: nvm

Comment (yuhinarita) [reply src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java:34]: If Skills.isValidSkill() already checks for empty skills, is this part necessary?

**Review by yuhinarita — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java (hunk 1 above)
Comment (ivan24004) [reply src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java:34]: This part checks for an empty skill set, `isValidSkill()`  will only check for a skill being empty in the set.

This ensures commands like `require-skill Group 1` are invalid as they do not mention any skills.

Comment (yuhinarita) [reply src/main/java/seedu/address/logic/parser/RequireSkillCommandParser.java:34]: Ok got it

**Pull request merged**